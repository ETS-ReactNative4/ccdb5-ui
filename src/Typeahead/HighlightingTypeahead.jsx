import React from 'react'
import PropTypes from 'prop-types'
import Typeahead from '.'

export const normalize = s => {
  return s.toLowerCase()
}

export const compileOptions = options => {
  return options.map(x => {
    return {
      key: x,
      normalized: normalize(x)
    }
  })
}

export default class HighlightingTypeahead extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      compiled: compileOptions(props.options)
    }
    this._onInputChange = this._onInputChange.bind(this)
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      compiled: compileOptions(nextProps.options)
    })
  }

  render() {
    return (
      <Typeahead {...this.props}
                 onInputChange={this._onInputChange}
                 renderOption={this._renderOption}
      />
    )
  }

  _onInputChange(value) {
    // Normalize the input value 
    const normalized = normalize(value)

    // Find the matches
    const filtered = this.state.compiled
      .filter(x => x.normalized.indexOf(normalized) !== -1)
      .map(x => {
        return {
          key: x.key,
          position: x.normalized.indexOf(normalized),
          value
        }
      })

    // Sort the matches so that matches at the beginning of the string
    // appear first
    filtered.sort((a,b) => {
      return a.position - b.position
    })

    return filtered
  }

  _renderOption(obj) {
    const start = obj.key.substring(0, obj.position)
    const match = obj.key.substr(obj.position, obj.value.length)
    const end = obj.key.substring(obj.position + obj.value.length)

    return {
      value: obj.key,
      component: (
        <span>{start}<b>{match}</b>{end}</span>
      )
    }
  }
}

HighlightingTypeahead.propTypes = {
  disabled: PropTypes.bool,
  maxVisible: PropTypes.number,
  minLength: PropTypes.number,
  onOptionSelected: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string
}

HighlightingTypeahead.defaultProps = {
  disabled: false,
  maxVisible: 5,
  minLength: 2,
  placeholder: 'Enter your search text',
  value: ''
}
