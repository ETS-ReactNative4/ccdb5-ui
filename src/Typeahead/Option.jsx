// Adapted from https://github.com/fmoo/react-typeahead

import React from 'react'
import PropTypes from 'prop-types'

export default class Option extends React.Component {
  render() {
    const classes = [ 'typeahead-option', 'body-copy' ]
    if ( this.props.selected ) {
      classes.push( 'selected' )
    }

    return (
     <li className={classes.join( ' ' )}
         onMouseDown={this.props.onClick}>
       { this.props.children }
     </li>
    )
  }
}

Option.propTypes = {
  selected: PropTypes.bool,
  onClick: PropTypes.func.isRequired
}

Option.defaultProps = {
  selected: false
}
