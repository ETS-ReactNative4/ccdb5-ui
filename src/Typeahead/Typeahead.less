@import (less) "../base.less";

.typeahead {
  padding-bottom: @gutter-normal;

  input {
    width: 100%;

    &:focus {
      border: solid 2px @pacific;
    }
  }

  .typeahead-selector {
    position: relative;

    z-index: 99;

    ul {
      border: solid 2px @pacific;
      border-top: 0;

      position: absolute;
      top: 0;
      right: 0;
      left: 0;

      li.footer {
        padding: @gutter-normal;
        border-top: solid 1px @gray-40;
        margin: 0;

        background-color: @gray-10;
        cursor: default;
        font-size: @size-vi;
        font-style: italic;
        list-style-type: none;
        text-align: center;
      }
    }
  }

  li.typeahead-option {
    padding: @gutter-normal;
    border-top: solid 1px @gray-40;
    margin: 0;

    background-color: @white;
    cursor: default;
    list-style-type: none;

    &:hover {
      border-top: solid 2px @pacific;
      border-bottom: solid 2px @pacific;

      background-color: @pacific-20;
    }

    &.selected {
      border-top: solid 2px @pacific;
      border-bottom: solid 2px @pacific;

      background-color: @pacific-20;
    }
  }
}
