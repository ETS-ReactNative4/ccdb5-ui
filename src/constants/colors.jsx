/* eslint-disable no-inline-comments, camelcase */
module.exports = Object.freeze( {
  BriteCharts: {
    regular: '#20aa3f',
    medium: '#ADDC91',
    light: '#C7E5B3'
  },
  DataLens: {
    0: '#2cb34a',
    1: '#addc91',
    2: '#257675',
    3: '#9ec4c3',
    4: '#0072ce',
    5: '#96c4ed',
    6: '#254b87',
    7: '#9daecc',
    8: '#b4267a',
    9: '#dc9cbf',
    10: '#a2a3a4'
  }
} )
