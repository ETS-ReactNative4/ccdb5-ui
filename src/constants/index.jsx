// Internal triggers
export const PAGE_CHANGED              = 'PAGE_CHANGED'
export const SEARCH_CHANGED            = 'SEARCH_CHANGED'
export const SIZE_CHANGED              = 'SIZE_CHANGED'
export const SORT_CHANGED              = 'SORT_CHANGED'
export const URL_CHANGED               = 'URL_CHANGED'
export const FILTER_CHANGED            = 'FILTER_CHANGED'
export const FILTER_REMOVED            = 'FILTER_REMOVED'
export const FILTER_ALL_REMOVED        = 'FILTER_ALL_REMOVED'

// External Triggers
export const COMPLAINTS_RECEIVED       = 'COMPLAINTS_RECEIVED'
export const COMPLAINTS_FAILED         = 'COMPLAINTS_FAILED'

// Useful constants
export const knownFilters = [
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tag',
  'timely',
  'zip_code'
]

export const SLUG_SEPARATOR            = 'â€¢'
