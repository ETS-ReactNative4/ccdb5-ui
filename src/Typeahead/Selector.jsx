// Adapted from https://github.com/fmoo/react-typeahead

import React from 'react'
import PropTypes from 'prop-types'
import Option from './Option'

export default class Selector extends React.Component {
  _onClick(result, event) {
    console.log(result)
  }

  render() {
    const results = this.props.options.map((x, i) => {
      const { component, value } = this.props.renderOption(x)
      return (
        <Option key={value + i}
                onClick={this._onClick.bind(this, value)}
                selected={this.props.selectedIndex === i}>
          {component}
        </Option>
      )
    })

    return (
      <div className="typeahead-selector">
        <ul>
          {results}
          { this.props.footer ? (<li className="footer">{this.props.footer}</li>) : null }
        </ul>
      </div>
    )
  }
}

Selector.propTypes = {
  footer: PropTypes.string,
  options: PropTypes.array,
  renderOption: PropTypes.func,
  selectedIndex: PropTypes.number  
}

Selector.defaultProps = {
  selectedIndex: -1
}