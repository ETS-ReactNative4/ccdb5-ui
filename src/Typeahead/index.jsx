// Adapted from https://github.com/fmoo/react-typeahead

import React from 'react'
import PropTypes from 'prop-types'
import Selector from './Selector'
import * as keys from '../constants'
import './Typeahead.less'

// ----------------------------------------------------------------------------
// Usage Mode

// The user can enter any text of choose one of the drop down options
export const MODE_OPEN = 'OPEN'

// The user is only allowed to choose one of the drop down options
export const MODE_CLOSED = 'CLOSED'

// ----------------------------------------------------------------------------
// State Machine

const ERROR = 'ERROR'
const EMPTY = 'EMPTY'
const ACCUM = 'ACCUM'
const WAITING = 'WAITING'
const NO_RESULTS = 'NO_RESULTS'
const RESULTS = 'RESULTS'
const TOO_MANY = 'TOO_MANY'
const CHOSEN = 'CHOSEN'

export const nextStateFromValue = (value, props) => {
  let phase = value ? WAITING : EMPTY
  if (value && value.length < props.minLength) {
    phase = ACCUM
  }

  return {
    inputValue: value,
    phase,
    searchResults: [],
    selectedIndex: -1
  }
}

export const nextStateFromOptions = (options, props) => {
  let phase = RESULTS
  if (!options || options.length === 0) {
    phase = NO_RESULTS
  }
  else if ( options && options.length > props.maxVisible) {
    phase = TOO_MANY
  }

  return {
    phase,
    searchResults: options,
    selectedIndex: -1
  }
}

// ----------------------------------------------------------------------------
// Class

export default class Typeahead extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      ...nextStateFromValue(props.value, props),
      focused: false
    }

    // Render/Phase Map
    this.renderMap = {
      ERROR: this._renderError.bind(this),
      EMPTY: this._renderEmpty.bind(this),
      ACCUM: this._renderEmpty.bind(this),
      WAITING: this._renderWaiting.bind(this),
      NO_RESULTS: this._renderNoResults.bind(this),
      RESULTS: this._renderResults.bind(this),
      TOO_MANY: this._renderTooManyResults.bind(this),
      CHOSEN: this._renderEmpty.bind(this)
    }

    // Key/function map
    this.keyMap = {}

    if (this.props.mode === MODE_OPEN ) {
      this.keyMap[keys.VK_ESCAPE] = this._openKeyCancel.bind(this)
      this.keyMap[keys.VK_UP] = this._openNav.bind(this, -1)
      this.keyMap[keys.VK_DOWN] = this._openNav.bind(this, 1)
      this.keyMap[keys.VK_ENTER] = this._chooseSelectedIndex.bind(this)
      this.keyMap[keys.VK_RETURN] = this._chooseSelectedIndex.bind(this)
      this.keyMap[keys.VK_TAB] = this._chooseSelectedIndex.bind(this)
    }
    else {
      this.keyMap[keys.VK_ESCAPE] = this._closedKeyCancel.bind(this)
      this.keyMap[keys.VK_UP] = this._closedNav.bind(this, -1)
      this.keyMap[keys.VK_DOWN] = this._closedNav.bind(this, 1)
      this.keyMap[keys.VK_ENTER] = this._chooseSelectedIndex.bind(this)
      this.keyMap[keys.VK_RETURN] = this._chooseSelectedIndex.bind(this)
      this.keyMap[keys.VK_TAB] = this._chooseSelectedIndex.bind(this)
    }

    // Bindings
    this._onBlur = this._onBlur.bind(this)
    this._onFocus = this._onFocus.bind(this)
    this._onKeyDown = this._onKeyDown.bind(this)
    this._onOptionsError = this._onOptionsError.bind(this)
    this._setOptions = this._setOptions.bind(this)
    this._valueUpdated = this._valueUpdated.bind(this)
  }

  componentWillReceiveProps(nextProps) {
    this.setState(nextStateFromValue(nextProps.value, nextProps))
  }

  componentDidUpdate() {
    this._callForOptions()
  }

  render() {
    return (
      <section className={`typeahead ${ this.props.className }`}
               onBlur={this._onBlur}
               onFocus={this._onFocus}>
        <input type="text"
               autoComplete="off"
               onChange={this._valueUpdated}
               onKeyDown={this._onKeyDown}
               placeholder={this.props.placeholder}
               value={this.state.inputValue}
               {...this.props.textBoxProps}
        />
        { this.state.focused ? this.renderMap[this.state.phase]() : null }
      </section>
    )
  }

  // --------------------------------------------------------------------------
  // Event Methods

  _onBlur(event) {
    this.setState({ focused: false })
  }

  _onFocus(event) {
    this.setState({ focused: true })
  }

  _onKeyDown(event) {
    const handler = this.keyMap[event.which]
    if (handler) {
      handler(event)
    }
  }

  _valueUpdated(event) {
    const nextState = nextStateFromValue(event.target.value, this.props)
    this.setState(nextState)
  }

  // --------------------------------------------------------------------------
  // Search Methods

  _calculateNewIndex(delta) {
    const max = Math.min(
      this.props.maxVisible, this.state.searchResults.length
    )
    if (max === 0) {
      return -1
    }

    // Clamp the new index
    let newIndex = this.state.selectedIndex + delta
    if (newIndex < 0) {
      newIndex = 0
    }
    if (newIndex >= max) {
      newIndex = max - 1
    }

    return newIndex
  }

  _callForOptions() {
    if (this.state.phase !== WAITING ) {
      return
    }

    const value = this.state.inputValue
    const returned = this.props.onInputChange(value)

    // Did the callback produce a promise?
    if (typeof(returned.then) === 'function') {
      returned.then(
        options => this._setOptions(options),
        error => this._onOptionsError(error)
      )
    }
    else {
      this._setOptions(returned)
    }
  }

  _onOptionsError() {
    this.setState({
      phase: ERROR
    })
  }

  _setOptions(options) {
    const nextState = nextStateFromOptions(options, this.props)
    this.setState(nextState)
  }

  _selectOption(index) {
    this.props.onOptionSelected(this.state.searchResults[index])
    const nextState = {
      phase: CHOSEN,
      searchResults: [],
      selectedIndex: -1
    }

    if (this.props.mode === MODE_CLOSED) {
      nextState.inputValue = ''
    }

    this.setState(nextState)
  }

  // --------------------------------------------------------------------------
  // Key Helpers

  _chooseSelectedIndex(event) {
    if (this.state.searchResults.length === 0) {
      return;
    }

    let idx = this.state.selectedIndex
    if (idx === -1) {
      idx = 0
    }

    this._selectOption(idx)
    event.preventDefault()
  }

  _closedKeyCancel(event) {
    event.preventDefault()
    this.setState(nextStateFromValue('', this.props))
  }

  _closedNav(delta, event) {
    event.preventDefault()
    const newIndex = this._calculateNewIndex(delta)
    if (newIndex >= 0) {
      this.setState({selectedIndex: newIndex})
    }
  }

  _openKeyCancel(event) {
    event.preventDefault()
    this.setState({phase: CHOSEN})
  }

  _openNav(delta, event) {
    event.preventDefault()
    const newIndex = this._calculateNewIndex(delta)
    if (newIndex >= 0) {
      // TODO: may need an extractor callback eventually
      // For now, assume the shape of the options
      const inputValue = this.state.searchResults[newIndex].key

      this.setState({
        selectedIndex: newIndex,
        inputValue
      })
    }
  }

  // --------------------------------------------------------------------------
  // Render Helpers

  _renderError() {
    return (<span className="error">There was a problem retrieving the options</span>)
  }

  _renderEmpty() {
    return null
  }

  _renderWaiting() {
    return (<span className="waiting">waiting...</span>)
  }

  _renderNoResults() {
    return (<span className="no-results">No results found</span>)
  }

  _renderResults() {
    return (
        <Selector options={this.state.searchResults}
                  onOptionSelected={this._selectOption.bind(this)}
                  renderOption={this.props.renderOption}
                  selectedIndex={this.state.selectedIndex}
        />
    )
  }

  _renderTooManyResults() {
    const subset = this.state.searchResults.slice(0, this.props.maxVisible)
    return (
        <Selector options={subset}
                  onOptionSelected={this._selectOption.bind(this)}
                  renderOption={this.props.renderOption}
                  selectedIndex={this.state.selectedIndex}
                  footer="Continue typing for more results"
        />
    )
  }
}

Typeahead.propTypes = {
  className: PropTypes.string,
  maxVisible: PropTypes.number,
  minLength: PropTypes.number,
  mode: PropTypes.oneOf([MODE_OPEN, MODE_CLOSED]).isRequired,
  onInputChange: PropTypes.func.isRequired,
  onOptionSelected: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  renderOption: PropTypes.func,
  textBoxProps: PropTypes.object,
  value: PropTypes.string
}

Typeahead.defaultProps = {
  className: '',
  maxVisible: 5,
  minLength: 2,
  mode: MODE_CLOSED,
  placeholder: 'Enter your search text',
  renderOption: (x) => {
    return {
      value: x,
      component: x
    }
  },
  textBoxProps: {},
  value: ''
}
