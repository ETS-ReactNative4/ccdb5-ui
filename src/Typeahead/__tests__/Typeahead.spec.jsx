import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import Typeahead from '..'
import * as keys from '../../constants'

function setupEnzyme(initalProps={}) {
  const props = Object.assign({
    onInputChange: jest.fn((x) => ['alpha', 'beta', 'gamma']),
    onOptionSelected: jest.fn(),
    renderOption: jest.fn()
  }, initalProps)

  const target = shallow(<Typeahead {...props} />);

  return {
    props,
    target
  }
}

function setupSnapshot(initialValue='') {
  return renderer.create(<Typeahead value={initialValue}
                                    onInputChange={jest.fn()}
                                    onOptionSelected={jest.fn()}
                         />)
}

describe('component::Typeahead', () => {
  describe('render phases', () => {
    it('renders the EMPTY phase', () => {
      const target = setupSnapshot()
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the ERROR phase', () => {
      const target = setupSnapshot()
      target.getInstance().setState({phase: 'ERROR'})
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the ACCUM phase', () => {
      const target = setupSnapshot('i')
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the WAITING phase', () => {
      const target = setupSnapshot('indubitably')
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the NO_RESULTS phase', () => {
      const target = setupSnapshot()
      target.getInstance()._setOptions([])
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the RESULTS phase', () => {
      const target = setupSnapshot()
      target.getInstance()._setOptions(['foo'])
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the TOO_MANY phase', () => {
      const target = setupSnapshot()
      target.getInstance()._setOptions(['foo', 'bar', 'baz', 'qaz', 'quux', 'nuux'])
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('focus/blur', () => {
    it('pushes the current state on blur', () => {
      const { target } = setupEnzyme({value: 'foo'})
      expect(target.state('phase')).toEqual('WAITING')
      target.simulate('blur')
      expect(target.state('phase')).toEqual('NOT_FOCUSED')
      expect(target.instance().stateHistory.length).toEqual(2)
    })

    it('restores the state on focus', () => {
      const { target } = setupEnzyme({value: 'foo'})
      expect(target.instance().stateHistory.length).toEqual(1)
      target.simulate('focus')
      expect(target.state('phase')).toEqual('WAITING')
      expect(target.instance().stateHistory.length).toEqual(0)
    })
  })

  describe('text control', () => {
    let target, props, input
    beforeEach(() => {
      ({target, props} = setupEnzyme())
      input = target.find('input')
    })

    it('requires a minimum number of characters before calling for options', () => {
      input.simulate('change', {target: { value: 'b'}})
      expect(target.state('inputValue')).toEqual('b')
      expect(props.onInputChange).not.toHaveBeenCalled()
    })

    it('calls for options when the threshhold has been reached', () => {
      input.simulate('change', {target: { value: 'bar'}})
      expect(target.state('inputValue')).toEqual('bar')
      expect(props.onInputChange).toHaveBeenCalledWith('bar')
    })
  })

  describe('keyboard events', () => {
    let fixture, target, props, input
    beforeEach(() => {
      ({target, props} = setupEnzyme())
      input = target.find('input')
      fixture = {
        preventDefault: jest.fn()
      }
    })

    it('ignores unknown keys', () => {
      fixture.which = 999
      input.simulate('keydown', fixture)
      expect(fixture.preventDefault).not.toHaveBeenCalled()
    })

    it('resets the control when "ESC" is pressed', () => {
      input.simulate('change', {target: { value: 'bar'}})
      expect(target.state('phase')).toEqual('RESULTS')
      
      fixture.which = keys.VK_ESCAPE
      input.simulate('keydown', fixture)

      expect(target.state('inputValue')).toEqual('')
      expect(target.state('phase')).toEqual('EMPTY')
      expect(fixture.preventDefault).toHaveBeenCalled()
    })

    describe('"DOWN" key', () => {
      it('has no effect when there are no options', () => {
        fixture.which = keys.VK_DOWN
        input.simulate('keydown', fixture)

        expect(target.state('selectedIndex')).toEqual(-1)
        expect(fixture.preventDefault).toHaveBeenCalled()
      })

      it('selects the first option when there is no selection', () => {
        input.simulate('change', {target: { value: 'bar'}})
        fixture.which = keys.VK_DOWN
        input.simulate('keydown', fixture)

        expect(target.state('selectedIndex')).toEqual(0)
      })

      it('has no effect when the last item is selected', () => {
        target.instance().setState({
          selectedIndex: 2,
          searchResults: ['foo', 'bar', 'baz']
        })

        fixture.which = keys.VK_DOWN
        input.simulate('keydown', fixture)

        expect(target.state('selectedIndex')).toEqual(2)
      })
    })

    describe('"UP" key', () => {
      it('has no effect when the first item is selected', () => {
        target.instance().setState({
          selectedIndex: 0,
          searchResults: ['foo', 'bar', 'baz']
        })

        fixture.which = keys.VK_UP
        input.simulate('keydown', fixture)

        expect(target.state('selectedIndex')).toEqual(0)
      })
    })

    describe('ENTER/TAB', () => {
      it('has no effect when there are no options', () => {
        fixture.which = keys.VK_ENTER
        input.simulate('keydown', fixture)
        expect(props.onOptionSelected).not.toHaveBeenCalled()
      })

      it('chooses the first option when there is no selection', () => {
        input.simulate('change', {target: { value: 'bar'}})
        fixture.which = keys.VK_TAB
        input.simulate('keydown', fixture)

        expect(props.onOptionSelected).toHaveBeenCalledWith('alpha')
      })

      it('selects the highlighted option when "TAB" is pressed', () => {
        input.simulate('change', {target: { value: 'bar'}})
        target.instance().setState({
          selectedIndex: 1
        })
        fixture.which = keys.VK_TAB
        input.simulate('keydown', fixture)

        expect(props.onOptionSelected).toHaveBeenCalledWith('beta')
      })
    })
  })
})
