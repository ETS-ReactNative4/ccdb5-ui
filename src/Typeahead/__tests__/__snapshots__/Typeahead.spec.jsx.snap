exports[`component::Typeahead render phases renders the ACCUM phase 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}>
  <input
    autoComplete="off"
    className="a-text-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter your search text"
    type="text"
    value="i" />
</section>
`;

exports[`component::Typeahead render phases renders the EMPTY phase 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}>
  <input
    autoComplete="off"
    className="a-text-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter your search text"
    type="text"
    value="" />
</section>
`;

exports[`component::Typeahead render phases renders the ERROR phase 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}>
  <input
    autoComplete="off"
    className="a-text-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter your search text"
    type="text"
    value="" />
  <span
    className="error">
    There was a problem retrieving the options
  </span>
</section>
`;

exports[`component::Typeahead render phases renders the NO_RESULTS phase 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}>
  <input
    autoComplete="off"
    className="a-text-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter your search text"
    type="text"
    value="" />
  <span
    className="no-results">
    No results found
  </span>
</section>
`;

exports[`component::Typeahead render phases renders the RESULTS phase 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}>
  <input
    autoComplete="off"
    className="a-text-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter your search text"
    type="text"
    value="" />
  <div
    className="typeahead-selector">
    <ul>
      <li
        className="typeahead-option body-copy"
        onMouseDown={[Function]}>
        foo
      </li>
    </ul>
  </div>
</section>
`;

exports[`component::Typeahead render phases renders the TOO_MANY phase 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}>
  <input
    autoComplete="off"
    className="a-text-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter your search text"
    type="text"
    value="" />
  <div
    className="typeahead-selector">
    <ul>
      <li
        className="typeahead-option body-copy"
        onMouseDown={[Function]}>
        foo
      </li>
      <li
        className="typeahead-option body-copy"
        onMouseDown={[Function]}>
        bar
      </li>
      <li
        className="typeahead-option body-copy"
        onMouseDown={[Function]}>
        baz
      </li>
      <li
        className="typeahead-option body-copy"
        onMouseDown={[Function]}>
        qaz
      </li>
      <li
        className="typeahead-option body-copy"
        onMouseDown={[Function]}>
        quux
      </li>
      <li
        className="footer">
        Continue typing for more results
      </li>
    </ul>
  </div>
</section>
`;
